### PR TITLE
Agency short names and filtering

### LINKED ISSUE #89: Design for Multi-agency search
- add system spec for any js used.

### PR DESCRIPTION
**Pull request by ondrae**
Uses the agency short names where we have them.
Screen Shot 2021-01-14 at 5 18 40 PM

Can filter on the long list of agency names. Does some clever js to still include checked but not visible agencies in the next search.

Screen Shot 2021-01-14 at 5 21 02 PM

Can also filter by short name.
Screen Shot 2021-01-14 at 5 20 41 PM

Closes #89 
Accessibility suggestions from #155 added.

Lots of tests included.

There was a different attempt at this in a closed PR #123

### CODE REVIEW

**Comment by ondrae** [comment]
Hmmm, I may having search be on its own route, and only load the js then.

**Review by peterrowland — CHANGES_REQUESTED** [review]
@iamjolly gave us some some [a11y recommendations](https://github.com/18F/all_sorns/issues/155) for this

**Comment by ondrae** [comment]
Added a label and the search filtering help text in a screen reader friendly way.
Screen Shot 2021-02-03 at 4 39 08 PM

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -16,7 +16,11 @@
 
     <%= stylesheet_link_tag 'application', media: 'all' %>
     <%= javascript_pack_tag 'application' %>
-    <%= javascript_pack_tag 'search' %>
+
+    <% if params[:search].present? %>
+      <!-- Only load cusotm js if there is an active search -->
```
Comment (ondrae) [inline app/views/layouts/application.html.erb:21]:
```suggestion
      <!-- Only load custom js if there is an active search -->
```

**Review by peterrowland — APPROVED** [review]

**Pull request merged**